### PR TITLE
Add cartesian sequence behavior

### PR DESCRIPTION
**Pull request by sjahr**
Depends on https://github.com/PickNikRobotics/moveit_studio/pull/4606

Add a small Cartesian motion demo that uses AddCartesianPoseToSequence and SetupMTCCartesianSequence

- [ ] Make it an infinite circle

### CODE REVIEW

File: src/picknik_ur_base_config/objectives/cartesian_sequence.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<root BTCPP_format="4" main_tree_to_execute="Cartesian Sequence">
+  <BehaviorTree ID="Cartesian Sequence" _description="Plan and execute a Cartesian path between a sequence of poses" _favorite="true">
+    <Control ID="Sequence" name="root">
+      <!-- Create demo waypoints -->
+      <!-- Waypoint 1 -->
+      <Action ID="CreateStampedPose" reference_frame="manual_grasp_link" position_xyz="-0.1;0.0;0.0" orientation_xyzw="0.0;0.0;0.0;1.0" stamped_pose="{stamped_pose}"/>
+      <Action ID="AddCartesianPoseToSequence" input_pose="{stamped_pose}" cartesian_sequence="{cartesian_sequence}"/>
```
Comment (sjahr) [inline src/picknik_ur_base_config/objectives/cartesian_sequence.xml:8]:
```suggestion
      <Action ID="AddPoseStampedToVector" input_pose="{stamped_pose}" pose_stamped_vector="{pose_stamped_vector}"/>
```

File: src/picknik_ur_base_config/objectives/cartesian_sequence.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      <!-- Waypoint 2 -->
+      <Action ID="CreateStampedPose" reference_frame="manual_grasp_link" position_xyz="-0.1;0.0;0.1" orientation_xyzw="0.0;0.0;0.0;1.0" stamped_pose="{stamped_pose}"/>
+      <Action ID="AddCartesianPoseToSequence" input_pose="{stamped_pose}" cartesian_sequence="{cartesian_sequence}"/>
```
Comment (sjahr) [inline src/picknik_ur_base_config/objectives/cartesian_sequence.xml:11]:
```suggestion
      <Action ID="AddPoseStampedToVector" input_pose="{stamped_pose}" pose_stamped_vector="{pose_stamped_vector}"/>
```

File: src/picknik_ur_base_config/objectives/cartesian_sequence.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      <!-- Waypoint 3 -->
+      <Action ID="CreateStampedPose" reference_frame="manual_grasp_link" position_xyz="0.0;0.0;0.1" orientation_xyzw="0.0;0.0;0.0;1.0" stamped_pose="{stamped_pose}"/>
+      <Action ID="AddCartesianPoseToSequence" input_pose="{stamped_pose}" cartesian_sequence="{cartesian_sequence}"/>
```
Comment (sjahr) [inline src/picknik_ur_base_config/objectives/cartesian_sequence.xml:14]:
```suggestion
      <Action ID="AddPoseStampedToVector" input_pose="{stamped_pose}" pose_stamped_vector="{pose_stamped_vector}"/>
```

File: src/picknik_ur_base_config/objectives/cartesian_sequence.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      <!-- Plan Cartesian motion -->
+      <Action ID="InitializeMTCTask" task_id="cartesian_sequence" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" task="{cartesian_sequence_task}"/>
+      <Action ID="SetupMTCCurrentState" task="{cartesian_sequence_task}"/>
+      <Action ID="SetupMTCCartesianSequence" name="SetupMTCCartesianSequence" planning_group_name="manipulator" cartesian_sequence="{cartesian_sequence}" ik_frame="manual_grasp_link" task="{cartesian_sequence_task}"/>
```
Comment (sjahr) [inline src/picknik_ur_base_config/objectives/cartesian_sequence.xml:18]:
```suggestion
      <Action ID="SetupMTCCartesianSequence" name="SetupMTCCartesianSequence" planning_group_name="manipulator" pose_stamped_vector="{pose_stamped_vector}" ik_frame="manual_grasp_link" task="{cartesian_sequence_task}"/>
```

File: src/picknik_ur_base_config/objectives/cartesian_sequence.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+      <Action ID="CreateStampedPose" reference_frame="manual_grasp_link" position_xyz="-0.1;0.0;0.0" orientation_xyzw="0.0;0.0;0.0;1.0" stamped_pose="{stamped_pose}"/>
+      <Action ID="AddPoseStampedToVector" input_pose="{stamped_pose}" pose_stamped_vector="{pose_stamped_vector}"/>
+      <!-- Waypoint 2 -->
+      <Action ID="CreateStampedPose" reference_frame="manual_grasp_link" position_xyz="-0.1;0.0;0.1" orientation_xyzw="0.0;0.0;0.0;1.0" stamped_pose="{stamped_pose}"/>
+      <Action ID="AddPoseStampedToVector" input_pose="{stamped_pose}" pose_stamped_vector="{pose_stamped_vector}"/>
+      <!-- Waypoint 3 -->
+      <Action ID="CreateStampedPose" reference_frame="manual_grasp_link" position_xyz="0.0;0.0;0.1" orientation_xyzw="0.0;0.0;0.0;1.0" stamped_pose="{stamped_pose}"/>
+      <Action ID="AddPoseStampedToVector" input_pose="{stamped_pose}" pose_stamped_vector="{pose_stamped_vector}"/>
+      <!-- Plan Cartesian motion -->
+      <Action ID="InitializeMTCTask" task_id="cartesian_sequence" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" task="{cartesian_sequence_task}"/>
```
Comment (MarqRazz) [inline src/picknik_ur_base_config/objectives/cartesian_sequence.xml:16]: This is done in a lot of tasks for some reason but why do we need the `/robotiq_gripper_controller` here?

File: src/picknik_ur_base_config/objectives/cartesian_sequence.xml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<root BTCPP_format="4" main_tree_to_execute="Cartesian Sequence">
+  <BehaviorTree ID="Cartesian Sequence" _description="Plan and execute a Cartesian path between a sequence of poses" _favorite="true">
```
Comment (MarqRazz) [inline src/picknik_ur_base_config/objectives/cartesian_sequence.xml:3]: Should this be a `favorite` Objective?

File: src/picknik_ur_base_config/objectives/cartesian_sequence.xml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+      <!-- Waypoint 2 -->
+      <Action ID="CreateStampedPose" reference_frame="manual_grasp_link" position_xyz="-0.1;0.0;0.1" orientation_xyzw="0.0;0.0;0.0;1.0" stamped_pose="{stamped_pose}"/>
+      <Action ID="AddPoseStampedToVector" input_pose="{stamped_pose}" pose_stamped_vector="{pose_stamped_vector}"/>
+      <!-- Waypoint 3 -->
+      <Action ID="CreateStampedPose" reference_frame="manual_grasp_link" position_xyz="0.0;0.0;0.1" orientation_xyzw="0.0;0.0;0.0;1.0" stamped_pose="{stamped_pose}"/>
+      <Action ID="AddPoseStampedToVector" input_pose="{stamped_pose}" pose_stamped_vector="{pose_stamped_vector}"/>
+      <!-- Plan Cartesian motion -->
+      <Action ID="InitializeMTCTask" task_id="cartesian_sequence" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" task="{cartesian_sequence_task}"/>
+      <Action ID="SetupMTCCurrentState" task="{cartesian_sequence_task}"/>
+      <Action ID="SetupMTCCartesianSequence" name="SetupMTCCartesianSequence" planning_group_name="manipulator" pose_stamped_vector="{pose_stamped_vector}" ik_frame="manual_grasp_link" task="{cartesian_sequence_task}"/>
```
[the remark this answers is not in the document]
Comment (MarqRazz) [reply src/picknik_ur_base_config/objectives/cartesian_sequence.xml:18]: What happens if they don't match?

**Comment by sjahr** [comment]
After coordination with @marioprats I am closing this in favor of a new objective that uses the behavior added by https://github.com/PickNikRobotics/moveit_studio/pull/4838

**Pull request closed without merging**